exaudfclient/docs/script_options_requirments.md

```diff
@@ -0,0 +1,124 @@
+# System Requirement Specification
+
+This document outlines the user-centric requirements for the new Exasol UDF Client Script Options parser. It identifies the key features, high-level requirements, and user roles to ensure clarity and alignment with user needs.
+
+## Roles
+
+This section lists the roles that will interact with or benefit from the parser system.
+
+### Database Administrator
+Database Administrators manage the database environment and ensure the efficient execution of UDFs within Exasol, configuring and overseeing script execution.
+
+### Data Scientist
+Data Scientists develop and deploy UDFs in languages such as Java, Python, or R to process and analyze data within Exasol.
+
+## Features
+
+This section lists the key features of the new UDF Client Script Options parser which you would highlight in a product leaflet.
+
+### General Script Options Parsing
+`feat~general-script-options-parsing~1`
+
+Needs: req
+
+### Java-specific Script Options
+`feat~java-specific-script-options~1`
+
+Needs: req
+
+## High-level Requirements
+
+This section details the high-level requirements for the new parser system, linked to the features listed above.
+
+### General Script Options Parsing
+`req~general-script-options-parsing~1`
+
+The parser must correctly identify and handle Script Options with the syntax `%optionKey optionValue;`. It must also manage whitespace and ignore non-Script Options as specified.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### Multiple and Duplicate Options Management
+`req~multiple-duplicate-options-management~1`
+
+The parser must collect multiple Script Options with the same key and handle duplicates according to specific rules for different options.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+- `feat~java-specific-script-options~1`
+
+### Script Option Removal
+`req~script-option-removal~1`
+
+The parser must remove found Script Options from the original script code.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+- `feat~java-specific-script-options~1`
+
+### Java %scriptclass Option Handling
+`req~java-scriptclass-option-handling~1`
+
+The parser must correctly identify a single %scriptclass option and remove any additional occurrences of this option within the script code.
+
+Needs: dsn
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %jar Option Handling
+`req~java-jar-option-handling~1`
+
+The parser must find multiple %jar options, handle duplicates properly, and ensure the values match the Java CLASSPATH environment variable syntax.
```

Comment:
what is with the order

Reply:
Order is defined in different requirements: `req~java-jar-option-handling-v1~1` and `req~java-jar-option-handling-v2~1`